index.bs

```diff
@@ -932,6 +932,159 @@ requests to start a WebDriver session, it must:
 
 </div>
 
+# Sandboxed Script Execution # {#sandbox}
+
+A common requirements for automation tools is to execute scripts which have
+access to the DOM of a document, but don't have information about any changes to
+the DOM APIs made by scripts running in the browsing context containing the
+document.
+
+A [=BiDi session=] has a <dfn>context sandbox map</dfn> which is a weak map
+in which the keys are [=browsing contexts=] and the values are weak maps between
+strings and {{SandboxWindowProxy}} objects.
+
+Note: The definition of sandboxes here is an attempt to codify the behaviour of
+existing implementations. It exposes parts of the implementations that have
+previously been considered internal by specifications, in particular the
+distinction between the internal state of platform objects (which is typically
+implemented as native objects in the main implementation language of the browser
+engine) and the ECMAScript-visible state. Because existing sandbox
+implementations happen at a low level in the engine, implementations converging
+toward the specficication in all details might be a slow process. In the
+meantime, implementors are encouraged to provide detailed documentation on any
+differences with the specification, and users of this feature are encouraged to
+explicitly test that scripts running in sandboxes work in all implementations.
+
+## Sandbox Realms ## {#sandbox-realm}
+
+Each sandbox is a unique EMCAScript [=Realm=]. However the sandbox realm
+provides access to platform objects in an existing {{Window}} realm via
+{{SandboxProxy}} objects.
+
+<div algorithm>
+To <dfn>get or create a sandbox realm</dfn> given |name| and |source browsing context|:
+
+1. If [=context sandbox map=] does not contain |name|, set |source browsing
```

Comment:
Looks like this algorithm doesn't quite align with the type of `context sandbox map` described above. Should this step check if `context sandbox map` contains `source browsing context` instead of `name` since it is a weak map whose keys are browsing contexts?